[READY]FLIGHTPACK PATCH 1: Crashing rework, drifting tweaks and fixes, DataHUDs, and slight price decrease!

silicons opened the pull request:
:cl: Mekhi Anderson
rscadd: Flightpacks are now easier to handle and use.
experimental: Flightpacks now respect directional momentum
experimental: Flightpacks now have a minimum speed before starting to drift
experimental: Flightsuit helmets now contain an omni-HUD
experimental: Flightpacks now check directions when crashing...
experimental: Flightpacks no longer instantly crash when crashing into living things
experimental: Flightpacks can now have their handling scrambled from EMP blasts
experimental: A flightpack crashing into the ground is now violent enough to disorient their user.
bugfix: Flightpacks can no longer multi-crash.
bugfix: Flightpack examining is now fixed. Cheers, TehZombehz!
experimental: Flightpacks now cost slightly less to make, but having the shoes is critical for handling...
\:cl:
@GunHog Modules/stock parts coming in a later update, or you could try to do them. Honk. Anyways this PR was long overdue as there's some fixes to unintentionally done things.

silicons committed:
pricedrop

silicons committed:
directionalcrashing

silicons committed:
..

silicons committed:
kek woops

silicons committed:
...

silicons committed:
changes

silicons committed:
HUDs

silicons committed:
woops

silicons committed:
huds...

silicons committed:
d

silicons committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into flightpack-patch1

silicons committed:
Update flightsuit.dm

GunHog commented on code/modules/research/designs/mechfabricator_designs.dm:
```diff
@@ -716,7 +716,7 @@
 	build_type = MECHFAB
 	build_path = /obj/item/clothing/suit/space/hardsuit/flightsuit
-	materials = list(MAT_METAL=20000, MAT_GLASS = 10000, MAT_DIAMOND = 200, MAT_GOLD = 3000, MAT_SILVER = 3000, MAT_TITANIUM = 20000)	//This expensive enough for you?
+	materials = list(MAT_METAL=16000, MAT_GLASS = 80000, MAT_DIAMOND = 200, MAT_GOLD = 3000, MAT_SILVER = 3000, MAT_TITANIUM = 16000)	//This expensive enough for you?
```
I am sure you forgot to remove a 0 for the glass cost.

silicons committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into flightpack-patch1

silicons committed:
Update mechfabricator_designs.dm

XDTM commented:
IMO, omnihud shouldn't be included in this. Omni-HUD glasses could be a late rnd item on their own.

kevinz000 commented:
Shadowlight's recommendation when I suggested letting them hold backpacks in sit storage

silicons committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into flightpack-patch1

# Conflicts:
#	code/modules/mob/living/carbon/human/human.dm

GunHog commented:
Looks like you botched your merge!

kevinz000 commented:
;-;

silicons committed:
.

silicons committed:
..
	new file:   config/lavaruinblacklist.txt
	new file:   config/spaceruinblacklist.txt
	new file:   config/unbuyableshuttles.txt

Shadowlight213 commented:
Omnihud seems like it fits thematically pretty well. Plus, it's an improvement that doesn't make the flight suit flat out better than integrated thrusters, since thrusters still have the advantage of having the back slot open.

kevinz000 commented:
thrusters also don't cost anything but are limited.
:honk:

silicons committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into flightpack-patch1

# Conflicts:
#	code/modules/clothing/spacesuits/flightsuit.dm

silicons committed:
.

AnturK commented on code/modules/clothing/spacesuits/flightsuit.dm:
```diff
@@ -439,93 +437,124 @@
+	else if(crashdir == EAST || crashdir == WEST)
+		V = "x"
+	var/crashpower = 0
+	switch(V)
```
This switch and V is redundant as far as i can tell. Just use the if blocks.

AnturK commented on code/modules/clothing/spacesuits/flightsuit.dm:
```diff
@@ -1139,14 +1176,26 @@
+	var/list/datahuds = list(DATA_HUD_SECURITY_ADVANCED, DATA_HUD_MEDICAL_ADVANCED, DATA_HUD_DIAGNOSTIC)
+
+/obj/item/clothing/head/helmet/space/hardsuit/flightsuit/equipped(mob/living/carbon/human/wearer, slot)
+	for(var/hudtype in datahuds)
```
Needs to call the parent.

AnturK commented on code/modules/clothing/spacesuits/flightsuit.dm:
```diff
@@ -1139,14 +1176,26 @@
+		H.add_hud_to(wearer)
+
+/obj/item/clothing/head/helmet/space/hardsuit/flightsuit/dropped(mob/living/carbon/human/wearer)
+	for(var/hudtype in datahuds)
```
Same as above

silicons committed:
fix

AnturK commented on code/modules/clothing/spacesuits/flightsuit.dm:
```diff
@@ -447,11 +447,10 @@
-			crashpower = momentum_speed_y
-		if("x")
-			crashpower = momentum_speed_x
+	if(V == "y")
```
I mean the whole intermidate V variable is redundant, it's not used anywhere but here so you might as well use the if(crashdir) blocks.

optimumtact commented on code/controllers/subsystem/timer.dm:
```diff
@@ -80,7 +80,7 @@ var/datum/subsystem/timer/SStimer
 	if(args.len > 4 || bad_args)
 		if(bad_args)
-			stack_trace("Invalid arguments in call to addtimer!")
+			stack_trace("Invalid arguments in call to addtimer! thingToCall:[thingToCall] procToCall:[procToCall] timeToRun:[world.time + wait].")
```
can you not

GunHog commented:
Please have a look at https://github.com/tgstation/tgstation/issues/21940 when you can.

silicons committed:
Update timer.dm

kevinz000 commented:
@optimumtact fixed :)

silicons committed:
Update flightsuit.dm

kevinz000 commented:
@GunHog should be fixed

RemieRichards commented on code/modules/clothing/spacesuits/flightsuit.dm:
```diff
@@ -439,93 +437,123 @@
-	var/mobonly = 0
-	if(crashing)	//We're already in the process of getting knocked around by a crash.
+	crashing = TRUE
+	var/V = ""
```
why not just use crashpower, instead of using this to find which crashpower to use?

kevinz000 replied:
Had the problem of people crashing when they clipped a mob perpendicular to their axis of movement, this fixes that far as I can tell...

RemieRichards replied:
I mean swap the V = Y and then checking if it's Y, with just setting crashpower under that check.
it's identical functionality, cleaner code.

silicons committed:
Update flightsuit.dm
